### DISCUSSION TITLE
What are the best practices for changing a theme token key?

### DISCUSSION THREAD

**Discussion by rwu823** [tailwindlabs/tailwindcss#12863] [Help]
```diff
/** @type {import('tailwindcss').Config} */
module.exports = {
    theme: {
      colors: {
-       'old-token': '#1fb6ff',
+       'new-token': 'new-color',
      },
    }
}
```

For example, we need to deprecate the old token. And many old tokens have already been used in mockups. Except from global search and replace, what else can I do? I have found [eslint-plugin-tailwindcss](https://github.com/francoismassart/eslint-plugin-tailwindcss), which can generate errors for elements using non-Tailwind classes. However, our project contains both Tailwind CSS and normal class names, so this error would not catch our attention.

**Comment by wongjn** [comment]
If `new-token` is a direct replacement for `old-token`, you could alias `old-token` to `new-token`:
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    colors: {
      'old-token': 'new-color',
      'new-token': 'new-color',
    },
  }
}
```
Otherwise, you could look at [creating your own ESLint plugin](https://eslint.org/docs/latest/extend/plugins) to catch the old token usage.

**Reply by rwu823** [reply]
@wongjn That sounds less practical. In some situations, we may need to change the `old-token` to a different color that better suits this naming. Moreover, having two keys representing the same token in the same codebase would be confusing.

**Reply by wongjn** [reply]
You could look at [creating your own ESLint plugin](https://eslint.org/docs/latest/extend/plugins) to catch the old token usage.
